pymc_marketing/clv/utils.py

```diff
@@ -887,42 +903,77 @@ def _expected_cumulative_transactions(
     date_range = pandas.date_range(start_date, periods=t + 1, freq=time_unit)
     date_periods = date_range.to_period(time_unit)
 
-    pred_cum_transactions = np.array([])
+    if model.purchase_covariate_cols or model.dropout_covariate_cols:
+        distinct_covariates_cols = list(
+            set(model.purchase_covariate_cols).intersection(
+                set(model.dropout_covariate_cols)
+            )
+        )
+        distinct_covariates = transactions[distinct_covariates_cols].drop_duplicates()
+    else:
+        distinct_covariates_cols = None
```

Comment:
There's a _customer_id_ dimension for the covariates in the fitted model `idata`. Rather than requiring covariates be added to be transaction dataframe, we can obtain the covariates from the model directly and merge them with `repeated_and_first_transactions`.